### DISCUSSION TITLE
16k VQGAN Mannequins

### DISCUSSION THREAD

**Discussion by afiaka87** [lucidrains/DALLE-pytorch#322] [Show and tell]
https://github.com/lucidrains/DALLE-pytorch/issues/319

https://www.dropbox.com/s/x9gbxiartrqaewf/afiaka_544I_96TSL_12D_16H_64DH_sparse.zip

Colab notebook for inference! 
https://colab.research.google.com/drive/11V2xw1eLPfZvzW8UQyTUhqCEU71w6Pr4?usp=sharing

Just reposting this here to close the issue

**Comment by afiaka87** [comment]
> @afiaka87 Works great!
> `a female mannequin dressed in a bright blue jacket and bright red skirt`
> ![samples](https://camo.githubusercontent.com/2d6ee0db9a807cdb7196405951d4f723469f2b97da0eeb98244fb595665f138f/68747470733a2f2f63646e2e646973636f72646170702e636f6d2f6174746163686d656e74732f3832333831333136303037353133323939312f3835383132323132333233343633393839322f71555549497551576f52796344396567456f426164487453696b3452616448713864693071716e704462536545454549494959515151676768684242434343474545454949495951515167676868424243434347456b49387535725962514167686842422e706e67)

**Comment by afiaka87** [comment]
a female mannequin dressed in an olive button-down shirt and gold palazzo pants
![a female mannequin dressed in an olive button-down shirt and gold palazzo pants](https://user-images.githubusercontent.com/3994972/123564776-4c404c00-d780-11eb-9c8e-3356df358df3.png)


an emoji of a baby penguin wearing a blue hat, blue gloves, red shirt, and green pants
![an emoji of a baby penguin wearing a blue hat, blue gloves, red shirt, and green pants](https://user-images.githubusercontent.com/3994972/123564838-7f82db00-d780-11eb-8e4f-98841acbbfab.png)

 a photo of westwood park, san francisco, from the water in the afternoon
![a photo of westwood park, san francisco, from the water in the afternoon](https://user-images.githubusercontent.com/3994972/123564891-b6f18780-d780-11eb-9019-8a1b6178f861.png)

a male mannequin dressed in an olive and black checkered button-down shirt and black pleated trousers
![a male mannequin dressed in an olive and black checkered button-down shirt and black pleated trousers](https://user-images.githubusercontent.com/3994972/123564961-e99b8000-d780-11eb-9384-fd228b90cb75.png)


a photo of the city streets of france
![a photo of the city streets of france](https://user-images.githubusercontent.com/3994972/123565245-da690200-d781-11eb-86c6-46c4c6b87f9c.png)

**Comment by soon-yau** [comment]
@afiaka87 I'm interested in training a model to generate mannequins, can you share the link of the dataset?

**Reply by afiaka87** [reply]
@soon-yau no problem - let me find it really quick.

**Reply by afiaka87** [reply]
@soon-yau 
Here's a link to the images:
https://www.dropbox.com/s/a4jx0pe6oc1e5r7/openai_dalle_gen.tar.gz 

and here are the captions:
https://www.dropbox.com/s/p0qwhefid4p8q0u/blog_captions.tar.gz

This contains the roughly 200K captioned pairs of mannequins wearing clothes, as well as around 800K captioned pairs for a variety of other situations.

```sh
mkdir Dataset
cd Dataset
wget --continue https://www.dropbox.com/s/p0qwhefid4p8q0u/blog_captions.tar.gz
wget --continue https://www.dropbox.com/s/a4jx0pe6oc1e5r7/openai_dalle_gen.tar.gz

tar xvf blog_captions.tar.gz
tar xvf openai_dalle_gen.tar.gz
```

**Reply by soon-yau** [reply]
Thanks @afiaka87 !!!!!!

**Reply by wangcong9299** [reply]
I am trying to download the link you shared, but it's no longer valid, can you share the link of the dataset again? Thank you very much.

**Reply by wangcong9299** [reply]
@afiaka87

**Comment by wangcong9299** [comment]
I am trying to download the link you shared, but it's no longer valid,  can you share the link of the dataset again?  Thank you very much.

**Reply by afiaka87** [reply]
Sure, let me find it real quick.

**Reply by soon-yau** [reply]
@afiaka87 Do you know the origin of the dataset e.g. dataset name, authors, paper, university etc? I'd like to cite that in a paper that I'm planning to write.

**Reply by afiaka87** [reply]
@wangcong9299 

These are the images:
https://www.dropbox.com/s/a4jx0pe6oc1e5r7/openai_dalle_gen.tar.gz

Captions:
https://www.dropbox.com/s/p0qwhefid4p8q0u/blog_captions.tar.gz?dl=0

To pair the captions with images, you will want to use the stem (e.g.`<stem>.png` or `<stem>.txt`) of the filename they share. Sorry they are a bit long.

**Reply by afiaka87** [reply]
@soon-yau The original images are all available from OpenAI's CDN on this specific blog post https://openai.com/blog/dall-e/. To scrape the full set of image caption pairs as I've done, you have to dig into the websites JavaScript. My understanding is that images provided from their CDN are licensed under creative commons, although you may want to double check this if it is relevant.

As for the authors, they are the authors of the DALLE paper and blog post. They can be found at  https://openai.com/blog/dall-e/ (at the bottom).

**Comment by peipeichenml** [comment]
Hi @afiaka87 I am trying to run your colab notebook, but I get this error:
```
Traceback (most recent call last):
  File "/content/dalle-pytorch-pretrained/DALLE-pytorch/generate.py", line 106, in <module>
    dalle.load_state_dict(weights)
  File "/usr/local/lib/python3.7/dist-packages/torch/nn/modules/module.py", line 1498, in load_state_dict
    self.__class__.__name__, "\n\t".join(error_msgs)))
RuntimeError: Error(s) in loading state_dict for DALLE:
	Missing key(s) in state_dict: "transformer.pos_emb", "transformer.layers.layers.0.0.fn.fn.fn.fn.fn.to_qkv.weight", "transformer.layers.layers.0.0.fn.fn.fn.fn.fn.to_out.0.weight", "transformer.layers.layers.0.0.fn.fn.fn.fn.fn.to_out.0.bias", "transformer.layers.layers.0.0.fn.fn.fn.fn.fn.attn_fn.master_layout", "transformer.layers.layers.0.1.fn.fn.fn.fn.net.0.weight", "transformer.layers.layers.0.1.fn.fn.fn.fn.net.0.bias", "transformer.layers.layers.0.1.fn.fn.fn.fn.net.3.weight", ...
```
Do you probably know what is wrong?
Thank you!

**Reply by afiaka87** [reply]
Not sure, this codebase and checkpoint are very old. I've moved on to GLIDE and Latent Diffusion, recommend the same.

**Reply by peipeichenml** [reply]
Thanks for replying. I will take a look at your recommendation :)
